src/hooks/useBestTrade.ts

```diff
@@ -32,16 +36,28 @@ export function useBestTrade(
     200
   )
 
+  const wethContract = useWETHContract()
+  const isAWrapTransaction = useMemo(() => {
+    if (!wethContract || !chainId || !amountSpecified || !debouncedOtherCurrency) return undefined
```

Comment:
Is it necessary to check if a `wethContract` exists, or is it enough to just check that the currency pair is native/weth?